syncer/syncer.go

```diff
@@ -1305,6 +1305,13 @@ func (s *Syncer) syncDML(
 			}
 		})
 
+		failpoint.Inject("SafeModeExit", func(val failpoint.Value) {
+			if intVal, ok := val.(int); ok && intVal == 4 {
+				s.tctx.L().Warn("fail to exec DML", zap.String("failpoint", "SafeModeExit"))
+				failpoint.Return(0, terror.ErrDBExecuteFailed.Delegate(errors.New("SafeModeExit"), "mock"))
+			}
+		})
+
```

Comment:
Actually I want to `executeSQLs` and then quit here. Directly return error here can't explain safe mode does take effect.